test/validation/api/atomic/atomic.c

```diff
@@ -954,6 +1054,93 @@ static void atomic_test_atomic_op_lock_free(void)
 	}
 }
 
+static void test_atomic_op_lock_free_128(void)
+{
+	odp_atomic_op_t atomic_op;
+	int ret_null, ret;
+
+	memset(&atomic_op, 0xff, sizeof(odp_atomic_op_t));
+	ret      = odp_atomic_lock_free_u128(&atomic_op);
+	ret_null = odp_atomic_lock_free_u128(NULL);
+
+	CU_ASSERT(ret == ret_null);
+
+	/* Init operation is not atomic by the spec. Call to
+	 * odp_atomic_lock_free_u128() zeros it but never sets it. */
+
+	if (ret == 0) {
+		/* none are lock free */
+		CU_ASSERT(atomic_op.all_bits     == 0);
+		CU_ASSERT(atomic_op.op.init      == 0);
+		CU_ASSERT(atomic_op.op.load      == 0);
+		CU_ASSERT(atomic_op.op.store     == 0);
+		CU_ASSERT(atomic_op.op.cas       == 0);
+	}
+
+	if (ret == 1) {
+		/* some are lock free */
+		CU_ASSERT(atomic_op.all_bits     != 0);
+		CU_ASSERT(atomic_op.op.cas       == 1);
```

Comment:
The CAS operation may not be always  lock-free, so this check should be removed.

Reply:
I will remove it in the next version. Thanks!